src/deploy/functions/validate.ts

```diff
@@ -35,9 +35,15 @@ export function functionsDirectoryExists(
  * @param functionNames Object containing function names as keys.
  * @throws { FirebaseError } Function names must be valid.
  */
-export function functionIdsAreValid(functions: { id: string }[]): void {
-  const validFunctionNameRegex = /^[a-zA-Z0-9_-]{1,63}$/;
-  const invalidIds = functions.filter((fn) => !validFunctionNameRegex.test(fn.id));
+export function functionIdsAreValid(functions: { id: string; platform: string }[]): void {
+  const v1FunctionName = /^[a-zA-Z][a-zA-Z0-9_-]{0,62}$/;
+  const v2FunctionName = /^[a-z][a-z-]{0,62}$/;
```

Comment:
```suggestion
  const v2FunctionName = /^[a-z][a-z0-9-]{0,62}$/;
```
I assume you can use numbers as well?

Reply:
Fixed.